pkg/org/auth.go

```diff
@@ -121,6 +137,55 @@ func UserFromRequest(app *bunapp.App, req bunrouter.Request) *bunconf.User {
 	return nil
 }
 
+func tryCloudflareFromRequest(app *bunapp.App, req bunrouter.Request) *bunconf.User {
+	if !app.Config().CloudflareAuthEnabled() {
+		return nil
+	}
+
+	// Adapted from https://developers.cloudflare.com/cloudflare-one/identity/authorization-cookie/validating-json/
+
+	headers := req.Header
+	accessJWT := headers.Get("Cf-Access-Jwt-Assertion")
+	if accessJWT == "" {
+		return nil
+	}
+
+	// TODO(aramperes): Initialize this on startup instead
+	var (
+		ctx        = req.Context()
+		teamDomain = app.Config().UserProviders.Cloudflare.TeamURL
+		certsURL   = fmt.Sprintf("%s/cdn-cgi/access/certs", teamDomain)
+		policyAUD  = app.Config().UserProviders.Cloudflare.Audience
+
+		config = &oidc.Config{
+			ClientID: policyAUD,
+		}
+		keySet   = oidc.NewRemoteKeySet(ctx, certsURL)
+		verifier = oidc.NewVerifier(teamDomain, keySet, config)
```

Comment:
Ideally this would be initialized once during startup. Note that the keys need to be refreshed every so often (I think once per month; I'm not sure if the library does that on its own).

@vmihailenco Any suggestions on how to make this fit in the codebase? Something in `App` or `Config`?

Reply:
I think we can refactor this code into a struct to make it stateful and then we can periodically reload the config. Let me send a PR.